core/src/Dscp/Resource/Class.hs

```diff
@@ -46,10 +48,15 @@ class AllocResource resource where
 
 -- | 'buildComponent' for 'ReaderT'.
 buildComponentR
-    :: Text
+    :: (HasLens' r LoggingIO)
+    => Text
     -> RIO r a
     -> (a -> RIO r ())
     -> ReaderT r ComponentM a
 buildComponentR desc allocate release =
     ReaderT $
-      \r -> buildComponent desc (runRIO r allocate) (runRIO r . release)
+      \r -> buildComponent desc
+                (runRIO r allocate)
```

Comment:
The original problem is the problem with DB authentication, which should arise during `allocate` - since `allocate` for Postgres includes schema preparation. It seems that if `boundTeardown` should be put around something, it more likely around `allocate`, not `release`.

Reply:
> DB authentication is handled fine, attempt to use non-existing user or incorrect password produces an error at `openPostgreSQL`.

But it never showed up in logs 🤔. As @flyingleafe, I’m not really sure how this PR fixes the issue. The behaviour I saw was that on failed authentication no error was reported by educator and it froze. Does it report the error with this fix applied?
I guess it would be really helpful if commits we make contained a description of the _cause_ of the original issue and how it is addressed in the commit, e.g. in this particular case how adding a timeout on _cleanup_ helps get the error reported.

Reply:
Oh, ok, I made it to the email in my inbox that has your YT comment and now everything is clear (I added my comment there). However, I’d like to reiterate that what you wrote in that YT comment should really have been in the commit message. (cc @gromakovsky)

Reply:
Right, this makes sense.